src/os.cc

```diff
@@ -41,6 +44,9 @@
 #    ifndef S_IROTH
 #      define S_IROTH 0
 #    endif
+#    ifndef mode_t
+#      define mode_t int
+#    endif
```

Comment:
This looks scary to me. What if `mode_t` is previously defined as a typedef, and you redefine it to a different type and never restore it.

Reply:
This is also scary if `mode_t` is actually a define.

Maybe have checks as
```
# if !defined(mode_t) && !defined(__mode_t_defined)
```

in both copies. I see the headers define `__mode_t_defined` macro when they typedef `mode_t`

Reply:
If mode_t is a macro this will be just a warning, no need to do anything.